### PR TITLE
MTDSA-5435 Template and Port

### PR DESCRIPTION
**Pull request by Nboaram**

### CODE REVIEW

File: app/definition/ApiDefinitionFactory.scala [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+          description = "Allow read access to self assessment data"
+        ),
+        Scope(
+          key = writeScope,
+          name = "Change your Self Assessment information",
+          description = "Allow write access to self assessment data"
+        )
+      ),
+      api = APIDefinition(
+        name = "#mtd-api# (MTD)",
```
Comment (james-work-account) [inline app/definition/ApiDefinitionFactory.scala:45]: Change this

File: app/definition/ApiDefinitionFactory.scala [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        description = "#desc#",
```
Comment (james-work-account) [inline app/definition/ApiDefinitionFactory.scala:46]: Change this

File: build.sbt [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    unmanagedClasspath in ItTest += baseDirectory.value / "resources",
+    unmanagedClasspath in Runtime += baseDirectory.value / "resources",
+    javaOptions in ItTest += "-Dlogger.resource=logback-test.xml",
+    parallelExecution in ItTest := false,
+    addTestReportOption(ItTest, "int-test-reports")
+  )
+  .settings(
+    resolvers += Resolver.jcenterRepo
+  )
+  .settings(PlayKeys.playDefaultPort := 7795) //TODO: Replace with desired default port for microservice.
```
Comment (james-work-account) [inline build.sbt:59]: Can you get rid of the TODO comment?

File: build.sbt [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+import sbt._
+import uk.gov.hmrc.DefaultBuildSettings.{addTestReportOption, defaultSettings}
+import uk.gov.hmrc.sbtdistributables.SbtDistributablesPlugin.publishingSettings
+import uk.gov.hmrc.versioning.SbtGitVersioning
+import uk.gov.hmrc.versioning.SbtGitVersioning.autoImport.majorVersion
+import uk.gov.hmrc.{SbtArtifactory, SbtAutoBuildPlugin}
+
+val appName = "mtd-api"
```
Comment (james-work-account) [inline build.sbt:24]: Change this

File: conf/application.conf [hunk 5]
```diff
@@ -0,0 +1,178 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This is the main configuration file for the application.
+# ~~~~~
+
+include "backend.conf"
+
+appName = mtd-api
```
Comment (james-work-account) [inline conf/application.conf:20]: Change this

File: conf/application.conf [hunk 6]
```diff
@@ -0,0 +1,178 @@
...
+
+appUrl = "http://localhost:#port#"
```
Comment (james-work-account) [inline conf/application.conf:22]: Change this

File: conf/application.conf [hunk 7]
```diff
@@ -0,0 +1,178 @@
...
+
+# Api related config
+api {
+  # The status of the version of the API for the API Platform.
+  1.0 {
+    status = "ALPHA"
+    endpoints.enabled = false
+  }
+  # The context which the API will have via the API Platform http://API_GATEWAY/{api.gateway.context}/
+  gateway.context = "mtd/template"
```
Comment (james-work-account) [inline conf/application.conf:113]: Change this

File: it/config/DocumentationControllerISpec.scala [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+package config
+
+import play.api.http.Status
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.WSResponse
+import support.IntegrationBaseSpec
+
+class DocumentationControllerISpec extends IntegrationBaseSpec {
+
+  val apiDefinitionJson: JsValue = Json.parse(
```
Comment (james-work-account) [inline it/config/DocumentationControllerISpec.scala:25]: This will need changing with the code/config changes

File: repository.yaml [hunk 9]
```diff
@@ -1,2 +1,2 @@
+digital-service: API Platform
```
Comment (james-work-account) [inline repository.yaml:1]: Where is this from? I would've thought that we'd want to keep MTD

File: resources/public/api/conf/1.0/application.raml [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+mediaType: application/hal+json
+
+uses:
+  sec: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/securitySchemes.raml
+  annotations: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/annotations.raml
+  types: https://developer.service.hmrc.gov.uk/api-documentation/assets/common/modules/types.raml
+  errors: errors.raml
+  headers: headers.raml
+
+'/mtd/template':
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/application.raml:32]: Change this

File: resources/public/api/conf/1.0/application.raml [hunk 11]
```diff
@@ -0,0 +1,48 @@
+#%RAML 1.0
+---
+title: MTD Template
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/application.raml:3]: Change this

File: resources/public/api/conf/1.0/docs/support.md [hunk 12]
```diff
@@ -0,0 +1,2 @@
+* Raise non-technical or platform-related issues with the [Software Development Support Team (SDST)](https://developer.service.hmrc.gov.uk/developer/support).
+* Raise technical issues on the [mtd-api](https://github.com/hmrc/mtd-api/issues) GitHub page.
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/docs/support.md:2]: Change this

File: resources/public/api/conf/1.0/example.html [hunk 13]
```diff
@@ -0,0 +1,239 @@
+@*
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/example.html:1]: We can get rid of this file

File: resources/public/api/conf/1.0/examples/example_1.json [hunk 14]
```diff
@@ -0,0 +1,16 @@
+{
+  "anIdUsedAcrossMultipleSchemas": "123456789012345",
+  "aFieldDefinedInline": "foo",
+  "links": [
+    {
+      "href": "/sample/context/AA11111A",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/examples/example_1.json:6]: Change this context

File: resources/public/api/conf/1.0/examples/example_1.json [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+      "rel": "self",
+      "method": "GET"
+    },
+    {
+      "href": "/sample/context/AA111111A/foo/bar",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/examples/example_1.json:11]: Change this context

File: resources/public/api/conf/1.0/examples/example_2.json [hunk 16]
```diff
@@ -0,0 +1,16 @@
+{
+  "anIdUsedAcrossMultipleSchemas": "123456789012346",
+  "aFieldDefinedInline": "bar",
+  "links": [
+    {
+      "href": "/insert/context/AA11112A",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/examples/example_2.json:6]: Change this context

File: resources/public/api/conf/1.0/examples/example_2.json [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+      "rel": "self",
+      "method": "GET"
+    },
+    {
+      "href": "/insert/context/AA111112A/foo/bar",
```
Comment (james-work-account) [inline resources/public/api/conf/1.0/examples/example_2.json:11]: Change this context

File: test/definition/ApiDefinitionFactorySpec.scala [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+import mocks.MockAppConfig
+import play.api.Configuration
+import support.UnitSpec
+import v1.mocks.MockHttpClient
+
+class ApiDefinitionFactorySpec extends UnitSpec {
+
+  class Test extends MockHttpClient with MockAppConfig {
+    val apiDefinitionFactory = new ApiDefinitionFactory(mockAppConfig)
+    MockedAppConfig.apiGatewayContext returns "mtd/template"
```
Comment (james-work-account) [inline test/definition/ApiDefinitionFactorySpec.scala:31]: Change this

File: test/definition/ApiDefinitionFactorySpec.scala [hunk 19]
```diff
@@ -0,0 +1,126 @@
...
+    "called" should {
+      "return a valid Definition case class" in new Test {
+        MockedAppConfig.featureSwitch returns None
+        MockedAppConfig.apiStatus returns "1.0"
+        MockedAppConfig.endpointsEnabled returns true
+
+        private val readScope = "read:self-assessment"
+        private val writeScope = "write:self-assessment"
+
+        apiDefinitionFactory.definition shouldBe
```
Comment (james-work-account) [inline test/definition/ApiDefinitionFactorySpec.scala:44]: This will need changing with the code/config changes

File: repository.yaml (hunk 9 above)
Comment (Nboaram) [reply repository.yaml:1]: dunno but it's in all the others

File: build.sbt [hunk 20]
```diff
@@ -0,0 +1,66 @@
...
+ */
+
+import sbt._
+import uk.gov.hmrc.DefaultBuildSettings.{addTestReportOption, defaultSettings}
+import uk.gov.hmrc.sbtdistributables.SbtDistributablesPlugin.publishingSettings
+import uk.gov.hmrc.versioning.SbtGitVersioning
+import uk.gov.hmrc.versioning.SbtGitVersioning.autoImport.majorVersion
+import uk.gov.hmrc.{SbtArtifactory, SbtAutoBuildPlugin}
+
+val appName = "individual-expenses-api"
```
Comment (Thepworth95) [inline build.sbt:24]: should this be individuals-expenses-api?

**Pull request merged**